cdap-spark-core-base/src/main/java/io/cdap/cdap/app/runtime/spark/classloader/SparkClassRewriter.java

```diff
@@ -468,17 +497,46 @@ public void onMethodExit(String name, String desc, GeneratorAdapter generatorAda
         generatorAdapter.push(STRUCTURED_RECORD_SERIALIZER_TYPE);
         generatorAdapter.invokeVirtual(KRYO_TYPE,
                                        new Method("addDefaultSerializer", Type.VOID_TYPE,
-                                                  new Type[] { Type.getType(Class.class), Type.getType(Class.class)}));
+                                                  new Type[] { Type.getType(Class.class), Type.getType(Class.class) }));
+
+        // Register serializer for all Unmodifiable classes within Collections. Ex:
+        // addDefaultSerializer(Collections.unmodifiableMap(new HashMap<>()).getClass(),
+        //                      UnmodifiableMapSerializer.class);
+        List<UnmodifiableData> unmodifableTypesData = new ArrayList<>();
```

Comment:
Use `Arrays.asList` to make the code simpler

```java
List<UnmodifiableData> unmodiableTypes = Arrays.asList(
  new UnmodifiableData("Collection", Collection.class, LinkedList.class),
  new UnmodifiableData("List", List.class, LinkedList.class),
  ...
);
```

Reply:
done